pilot/pkg/security/authz/model/generator.go

```diff
@@ -181,8 +181,17 @@ func (requestPresenterGenerator) principal(key, value string, forTCP bool) (*rba
 
 type requestHeaderGenerator struct{}
 
-func (requestHeaderGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
-	return nil, fmt.Errorf("unimplemented")
+func (requestHeaderGenerator) permission(key, value string, forTCP bool) (*rbacpb.Permission, error) {
```

Comment:
does this impact non-ambient cases?

Reply:
No, the header generator is added in non-ambient under the prinicipals rules.